runtime/parachains/src/paras_inherent.rs

```diff
@@ -355,11 +350,12 @@ pub mod pallet {
 					.map(|s| (s.session, s.candidate_hash))
 					.collect();
 
+				// Note that `provide_multi_dispute_data` will iterate, verify, and import each
+				// dispute; so the input here must be reasonably bounded.
 				let _ = T::DisputesHandler::provide_multi_dispute_data(disputes.clone())?;
 				if T::DisputesHandler::is_frozen() {
 					// The relay chain we are currently on is invalid. Proceed no further on parachains.
-					Included::<T>::set(Some(()));
-					return Ok(Some(minimal_inherent_weight::<T>()).into())
+					return Ok(Some(dispute_statements_weight::<T>(&disputes)).into())
```

Comment:
If this does not get merged today I will pull out this change to its own  mini PR since its not related to the other changes

Reply:
https://github.com/paritytech/polkadot/pull/4281